Mathlib/RingTheory/WittVector/Basic.lean

```diff
@@ -62,235 +62,216 @@ variable {α : Type _} {β : Type _}
 
 -- mathport name: expr𝕎
 local notation "𝕎" => WittVector p
+local notation "W_" => wittPolynomial p
 
 -- type as `\bbW`
 open scoped Witt
 
 namespace WittVector
 
 /-- `f : α → β` induces a map from `𝕎 α` to `𝕎 β` by applying `f` componentwise.
-If `f` is a ring homomorphism, then so is `f`, see `witt_vector.map f`. -/
+If `f` is a ring homomorphism, then so is `f`, see `WittVector.map f`. -/
 def mapFun (f : α → β) : 𝕎 α → 𝕎 β := fun x => mk _ (f ∘ x.coeff)
 #align witt_vector.map_fun WittVector.mapFun
 
-namespace MapFun
+namespace mapFun
 
-theorem injective (f : α → β) (hf : Injective f) : Injective (mapFun f : 𝕎 α → 𝕎 β) := fun x y h =>
-  ext fun n => hf (congr_arg (fun x => coeff x n) h : _)
+-- porting note: switched the proof to tactic mode. I think that `ext` was the issue.
+theorem injective (f : α → β) (hf : Injective f) : Injective (mapFun f : 𝕎 α → 𝕎 β) := by
+  intros _ _ h
+  ext p
+  exact hf (congr_arg (fun x => coeff x p) h : _)
 #align witt_vector.map_fun.injective WittVector.mapFun.injective
 
 theorem surjective (f : α → β) (hf : Surjective f) : Surjective (mapFun f : 𝕎 α → 𝕎 β) := fun x =>
-  ⟨mk _ fun n => Classical.choose <| hf <| x.coeff n, by ext n; dsimp [map_fun];
-    rw [Classical.choose_spec (hf (x.coeff n))]⟩
+  ⟨mk _ fun n => Classical.choose <| hf <| x.coeff n,
+    by ext n; simp only [mapFun, coeff_mk, comp_apply, Classical.choose_spec (hf (x.coeff n))]⟩
 #align witt_vector.map_fun.surjective WittVector.mapFun.surjective
 
-variable (f : R →+* S) (x y : 𝕎 R)
-
-/- ./././Mathport/Syntax/Translate/Expr.lean:330:4: warning: unsupported (TODO): `[tacs] -/
-/-- Auxiliary tactic for showing that `map_fun` respects the ring operations. -/
-unsafe def map_fun_tac : tactic Unit :=
-  sorry
-#align witt_vector.map_fun.map_fun_tac witt_vector.map_fun.map_fun_tac
-
-include hp
-
-/- ./././Mathport/Syntax/Translate/Tactic/Builtin.lean:69:18: unsupported non-interactive tactic witt_vector.map_fun.map_fun_tac -/
+-- porting note: using `(x y : 𝕎 R)` instead of `(x y : WittVector p R)` produced sorries.
+variable (f : R →+* S) (x y : WittVector p R)
+
+/-- Auxiliary tactic for showing that `mapFun` respects the ring operations. -/
+--  porting note: a very crude port.
+macro "map_fun_tac" : tactic => `(tactic| (
+  ext n
+  simp only [mapFun, mk, comp_apply, zero_coeff, map_zero,
+    -- porting note: the lemmas on the next line do not have the `simp` tag in mathlib4
+    add_coeff, sub_coeff, mul_coeff, neg_coeff, nsmul_coeff, zsmul_coeff, pow_coeff,
+    peval, map_aeval, algebraMap_int_eq, coe_eval₂Hom]  <;>
+  try { cases n <;> simp <;> done }  <;>  -- porting note: this line solves `one`
+  apply eval₂Hom_congr (RingHom.ext_int _ _) _ rfl  <;>
+  ext ⟨i, k⟩ <;>
+    fin_cases i <;> rfl ))
+
+--  porting note: mathport left a long line, probably due to the presence of meta code, from here
```

Comment:
```suggestion
```
yeah those lines should just be removed, and we don't even leave porting notes about them (-;